pages/posts/website-analytics-snapshot-2023.md

```diff
@@ -0,0 +1,171 @@
+---
+title: "50+ million requests in 7 days on json-schema.org"
+date: 2023-07-31T06:00:00+00:00
+type: engineering
+cover: /img/posts/2023/website-analytics-snapshot-2023/cover.webp
+authors:
+  - name: Ben Hutton
+    photo: /img/avatars/benhutton.webp
+    link: https://twitter.com/relequestual
+excerpt: "See what we discovered when looking at over 50 million requests to the JSON Schema org domain from the last 7 days"
+---
+
+I grant you, 7 days isn't an ideal length of time to do an analysis over, but this is what we have for now. Previously, we had basic website analytics, but that only tells a fraction of the story. We run the website on GitHub pages with Cloudflare infront, which provides 7 days of analytics, critically, including direct access traffic.
+
+The JSON Schema website hosts the official meta-schemas. If you're not sure what that means, meta-schemas are JSON Schemas for JSON Schema. Why is this interesting? Well, we host all historical versions of the meta-schema, not just the most recent. And while we don't expect code to be making requests on a regular basis to download the meta-schemas, the number of requests strongly suggests that they do.
+
+Why now? In addition to providing meta-schemas at versioned URLs, the web site has, and continues to, provide the latest meta-schema at `/schema`. We currently advise people not to be accessing this! We put in a delayed redirect with a message warning people to avoid it, but realistically, we expect most access will be via code and not in a browser.
+
+Maybe now we can remove hosting that URL? For reasons out of the scope of this article, using it really isn't a great idea, and will probably produce results you don't expect. As mentioned, web analytics won't tell us what's happening outside of a browser, so we needed something else… We were already using Cloudflare, so decided to submit for the open source free pro tier plan… while we wait, the cost isn't huge, so I decided we should look at the data now.
+
+The data in this article is a 7 day snapshot taken at 2023-07-21.
+(At the end of the article, I'll share an image from this snapshot, and one from 2023-07-31 for comparison.)
+
```

Comment:
Maybe applying some markdown styling to the caveats list will improve readability.